src/node/src/client.js

```diff
@@ -551,6 +551,47 @@ exports.makeClientConstructor = function(methods, serviceName) {
     this.updateMetadata = updateMetadata;
   }
 
+  /**
+   * Wait for the client to be ready. The callback will be called when the
+   * client has successfully connected to the server, and it will be called
+   * with an error if the attempt to connect to the server has unrecoverablly
+   * failed or if the deadline expires. This function does not automatically
+   * attempt to initiate the connection, so the callback will not be called
+   * unless you also start a method call or call $tryConnect.
+   * @param {(Date|Number)} deadline When to stop waiting for a connection. Pass
+   *     Infinity to wait forever.
+   * @param {function(Error)} callback The callback to call when done attempting
+   *     to connect.
+   */
+  Client.prototype.$waitForReady = function(deadline, callback) {
+    var self = this;
+    var checkState = function(err, result) {
+      if (err) {
+        callback(new Error('Failed to connect before the deadline'));
+      }
+      var new_state = result.new_state;
+      console.log(result);
+      if (new_state === grpc.connectivityState.READY) {
+        callback();
+      } else if (new_state === grpc.connectivityState.FATAL_FAILURE) {
+        callback(new Error('Failed to connect to server'));
+      } else {
+        self.channel.watchConnectivityState(new_state, deadline, checkState);
+      }
+    };
+    checkState(null, {new_state: this.channel.getConnectivityState()});
```

Comment:
I wasn't really sure whether it would be more intuitive to have it initiate the connection or just alert when there is a connection.